design/one-pager-stack-cli.md

```diff
@@ -0,0 +1,79 @@
+# Crossplane Stacks CLI
+* Owner: Daniel Suskin (@suskin)
+* Reviewers: Crossplane Maintainers
+* Status: Draft
+
+## Introduction
+*Why does this document exist?*
+
+We're working on the first version of a user experience for working with crossplane stacks. This document answers the questions of:
+* What are our goals for this version?
+* How do we measure success?
+* What do we think the experience will look like?
+
+## OKRs
+*What are our goals for this version? How do we measure success?*
+
+Objectives:
+* Provide a complete and simple experience for creating stacks from scratch (for a single pattern of creating stacks)
+* The stacks experience can be updated independently of the Crossplane releases
+
+Key Results:
+* Creating a Hello World stack from scratch takes 5 minutes, using the recommended approach
+* Creating and publishing a Hello World stack from scratch takes 10 minutes, using the recommended approach
+* All documentation linking from Crossplane 0.3 to the stacks experience will not need to be changed if we update the stacks experience
+
+## Experience
+*What do we think the experience will look like?*
+
+To create a Hello World stack from scratch:
+* Have access to a Crossplane
+* Install kubebuilder v2 and its prerequisites
+  - go
+  - docker
+  - kustomize (optional)
+* Create a kubebuilder v2 project
+* Add a "Hello World" log line to the kubebuilder controller
+* Install kubectl plugins
+  - `curl -s -o /usr/local/bin/kubectl-crossplane-stack-init /url/directly/to/the/script/somewhere >/dev/null`
+  - More curls until all the scripts are downloaded
+  - `chmod +x /usr/local/bin/kubectl-crossplane-*`
+* Navigate to the project directory
+  - `cd myproject`
+* Run init
+  - `kubectl crossplane stack init mystackgroup/mystackname`
+* Run build
+  - `kubectl crossplane stack build`
+* Run publish
+  - (docker registry credentials should be set up before publishing)
+  - `kubectl crossplane stack publish`
+* Install
```

Comment:
what about an install experience for stacks that this tooling did not init and build?

e.g. `kubectl crossplane stack install crossplane/stack-gcp`

Reply:
Good point . . . originally I was not imagining this as part of the initial toolset, but since we've added it to the scope of the initial toolset, it probably makes sense to mention it here too.

Reply:
This should be addressed now, so I'm going to resolve the conversation